### PR TITLE
Feature profile

### PR DESCRIPTION
**Pull request by Anas12091101**

### CODE REVIEW

File: ScratchDjango/Otp/models.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models import CharField
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+
+# @receiver(pre_save, sender=User)
```
Comment (zamanafzal) [inline ScratchDjango/Otp/models.py:7]: Please remove this code

File: ScratchDjango/Otp/urls.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from django.urls import path
+
+from .views import check_otp, create_otp
+
+urlpatterns = [path("check/",check_otp, name="check_otp"),
```
Comment (zamanafzal) [inline ScratchDjango/Otp/urls.py:5]: can we make url more descriptive as to the name of view?

File: ScratchDjango/Otp/utils.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        mailto,
+        reply_to=[EMAIL_HOST_USER],
+    )
+    email.send(fail_silently=False)
+
+
+def generate_otp(user):
+    otp_base32 = pyotp.random_base32()
+    otp_auth_url = pyotp.totp.TOTP(otp_base32).provisioning_uri(
+        name=user.email.lower(), issuer_name="scratchdjango"
```
Comment (zamanafzal) [inline ScratchDjango/Otp/utils.py:48]: can we pick issuer_name  from settings?

File: ScratchDjango/Profile/serializers.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+from ScratchDjango.User.serializers import UserSerializer
+
+from .models import Profile
+from .validators import validate_image
+
+
+class ProfileSerializer(serializers.Serializer):
+    user = UserSerializer()
+    avatar = serializers.ImageField(validators=[validate_image])
+    bio = serializers.CharField()
```
Comment (zamanafzal) [inline ScratchDjango/Profile/serializers.py:13]: Good to see you applied validation here.

File: ScratchDjango/Profile/serializers.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    
+    def create(self, validated_data):
```
Comment (zamanafzal) [inline ScratchDjango/Profile/serializers.py:15]: Can we do this using signal? whenenver a user created his profile is created automatically?

File: ScratchDjango/Profile/views.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+from .models import Profile
+from .serializers import ProfileSerializer
+
+
+# Create your views here.
+@api_view(["POST","PUT"])
+@permission_classes([IsAuthenticated])
+def create_or_update(request):
+    try:
```
[the remark this answers is not in the document]
Comment (Anas12091101) [reply ScratchDjango/Profile/views.py:18]: I first used it for common exception occurring at the time of coding, just forgot to remove it

**Pull request merged**